internal/crypto/cert.go

```diff
@@ -59,3 +65,71 @@ func ParseCertificates(data []byte, contentType string) (certs []*x509.Certifica
 	}
 	return parsePEM(data)
 }
+
+// MergeCertificateChain is a function that takes in a plugin configuration map
+// and a slice of x509 certificate instances, and attempts to merge the
+// certificate chain of the original certificates with a certificate bundle
+// specified in the plugin configuration.
+func MergeCertificateChain(certBundlePath string, originalCerts []*x509.Certificate) ([]*x509.Certificate, error) {
+	// read certificate bundle
+	certBundleBytes, err := os.ReadFile(certBundlePath)
+	if err != nil {
+		return nil, fmt.Errorf("cannot open user provided cert bundle file %q with err: %w", certBundlePath, err)
+	}
+	certBundle, err := parsePEM(certBundleBytes)
+	if err != nil {
+		return nil, fmt.Errorf("cannot parse user provided cert bundle file %q with err: %w", certBundlePath, err)
+	}
+	if len(certBundle) == 0 {
+		return nil, errors.New("the certificate bundle file is empty or not in PEM format")
+	}
+
+	// merge certificate chain
+	return mergeCertificateChain(originalCerts, certBundle)
```

Comment:
how about:
```suggestion
	return ValidateCertificateChain(append(originalCerts, certBundle...))
```
and we can remove `mergeCertificateChain`.

Reply:
Updated.